src/test/java/org/checkstyle/autofix/recipe/AbstractRecipeTestSupport.java

```diff
@@ -105,34 +123,50 @@ private List<CheckstyleViolation> runCheckstyle(String inputPath,
         checker.process(filesToCheck);
 
         final Path tempXmlPath = Files.createTempFile("checkstyle-report", ".xml");
-        try {
-            Files.write(tempXmlPath, xmlOutput.toByteArray());
-            return CheckstyleReportParser.parse(tempXmlPath);
-        }
-        finally {
-            Files.deleteIfExists(tempXmlPath);
-        }
+
+        Files.write(tempXmlPath, xmlOutput.toByteArray());
+        return tempXmlPath;
     }
 
-    private void verifyOutputFile(String outputPath, Configuration config) throws Exception {
+    private Path createConfigFile(Configuration config) throws Exception {
+        final CheckConfiguration checkConfig = ConfigurationLoader.mapConfiguration(config);
+        final Path configPath = Files.createTempFile("checkstyle-config", ".xml");
+        writeConfigurationToXml(checkConfig, configPath.toString());
+        return configPath;
+    }
 
-        final List<CheckstyleViolation> violations = runCheckstyle(outputPath, config);
-        if (!violations.isEmpty()) {
-            final StringBuilder violationMessage =
-                    new StringBuilder("Checkstyle violations found in the output file:\n");
+    public static void writeConfigurationToXml(CheckConfiguration config, String filePath)
```

Comment:
We have access to input file, why can't we extract the config part and put it into a file?

Reply:
have you tried to add it? what issues did you face with?